src/App.tsx

```diff
@@ -1,14 +1,149 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { USER_ID, createTodo, getTodos, updateTodo } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+type SortSelectOptions = 'all' | 'active' | 'completed';
+export type ErrorMessages =
+  | ''
+  | 'Unable to load todos'
+  | 'Title should not be empty'
+  | 'Unable to add a todo'
+  | 'Unable to delete a todo'
+  | 'Unable to update a todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectSort, setSelectSort] = useState<SortSelectOptions>('all');
+  const [error, setError] = useState<ErrorMessages>('');
+  const [inputTodo, setInputTodo] = useState('');
+  const [loading, setLoading] = useState(false);
+
+  const todoList = todos.filter(todo => {
+    if (selectSort === 'active') {
+      return !todo.completed;
+    } else if (selectSort === 'completed') {
+      return todo.completed;
+    }
+
+    return true;
+  });
+
+  useEffect(() => {
+    setLoading(true);
+
+    getTodos()
+      .then(todosPromise => {
+        setTodos(todosPromise);
+      })
+      .catch(() => setError('Unable to load todos'))
+      .finally(() => setLoading(false));
+  }, []);
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      setError('');
+    }, 3000);
+
+    return () => clearTimeout(timer);
+  }, [error]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  function handleKeyDown(event: React.KeyboardEvent<HTMLInputElement>) {
+    setError('');
+    if (event.key === 'Enter') {
```

Comment:
Use early return instead of having your entire logic inside a conditional:

Instead of
```js
if(neededCondition) { 
    ...
}
```

Do:
```js
if(!neededCondition) return;
...
```